### PR TITLE
Font close second attempt

### PR DESCRIPTION
**Pull request by typemytype**

### CODE REVIEW

**Comment by anthrotype** [comment]
we had a similar issue in booleanOperations, where appveyor tests were failing because pip was outdated inside the venv created by tox.
try to this:
https://github.com/typemytype/booleanOperations/commit/8a4d0ecb62cd8d71893cc096a4ff9b6922d823f1

**Comment by justvanrossum** [comment]
Hm, this PR needs to be rebased. The PR branch is on the main repo, so not on a fork. @anthrotype, can you help me doing the Right Thing? I'm a little traumatized after my last PR rebase disaster... Or anyone else know the correct git incantations?

**Comment by justvanrossum** [comment]
On second though, there are some changes in this PR that actually conflict with the recent ufoFormatVersion changes. Let me see.

**Comment by justvanrossum** [comment]
Whoohoo, tests pass! I think this is mergable.

**Comment by benkiel** [comment]
Ummm, @justvanrossum this is awesome! Thank you!

File: Lib/defcon/objects/font.py [hunk 1]
```diff
@@ -188,6 +187,21 @@ def __init__(self, path=None,
             layer = self.newLayer("public.default")
             self._layers.defaultLayer = layer
 
+    def close(self):
+        # close file systems
+        if self._reader is not None:
+            self._reader.close()
+            del self._reader
```
Comment (anthrotype) [inline Lib/defcon/objects/font.py:194]: why `del` and not setting the attribute back to `None`?

**Pull request merged**